rpcperms/interceptor.go

```diff
@@ -0,0 +1,256 @@
+package rpcperms
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btclog"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/lightningnetwork/lnd/monitoring"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+// RPCInterceptor is a struct that can be added to the running GRPC server,
+// intercepting API calls. This is useful for logging, enforcing permissions
+// etc.
+type RPCInterceptor struct {
```

Comment:
Could we call this `InterceptorChain`? I'm working on something that'll build on top of this and will actually intercept (and potentially manipulate) RPC messages, similar with what we do in the HTLC interceptor (need this for the macaroon accounts feature). So I would also call that `RPCInterceptor`.
I was also trying to come up with a good and short package name. I like `rpcperms` :+1:

Reply:
I like it, done!